coc/static/update_static.py

```diff
@@ -6,229 +6,1482 @@
-                # remove from base_level
-                del levels_dict[base_level][col]
-
-    # 4) Write final JSON
-    import json
-    with open(f"{save_name}.json", "w", encoding="utf-8") as jf:
-        jf.write(json.dumps(final_data, indent=2))
-
-
-def check_header(data):
-    if data[0] == 0x5D:
-        return "csv"
-    if data[:2] == b"\x53\x43":
-        return "sc"
-    if data[:4] == b"\x53\x69\x67\x3a":
-        return "sig:"
-    if data[:6] == b'"Name"' or data[:6] == b'"name"':
-        return "decoded csv"
-    raise Exception("Unknown header")
-
-def main():
-    for target_file, target_save in TARGETS:
-        target_save = target_file if target_save is None else target_save
-        download_url = f"{BASE_URL}/{target_file}"
-
-        print(f"Downloading: {download_url}")
-        with urllib.request.urlopen(download_url, timeout=10) as conn:
-            data = conn.read()
-
-        # Save raw compressed data
-        with open(target_save, "wb") as f:
-            f.write(data)
 
-        file_type = check_header(data)
-        if file_type == "csv":
-            # data is already uncompressed CSV
-            process_csv(data=data, file_path=target_save, save_name=target_save.split(".")[0])
-        elif file_type == "sig:":
-            # skip first 68 bytes, then decompress
-            process_csv(data=data[68:], file_path=target_save, save_name=target_save.split(".")[0])
-        else:
-            # treat as compressed
-            process_csv(data=data, file_path=target_save, save_name=target_save.split(".")[0])
+            #for merged buildings, move the requirement to level 1 since that is when the requirement is actually needed
+            if building_data.get("MergeRequirement") is not None:
+                building_data["1"]["MergeRequirement"] = building_data.get("MergeRequirement")
+
+            hold_data = {
+                "_id": _id,
+                "name": self._translate(tid=building_data.get("TID")),
+                "info": self._translate(tid=building_data.get("InfoTID")),
+                "TID": {
+                    "name": building_data.get("TID"),
+                    "info": building_data.get("InfoTID"),
+                },
+                "type": building_data.get("BuildingClass"),
+                # we are going to do this purely for builder hut purposes bc lv 1 is gems
+                "upgrade_resource": self._parse_resource(
+                    resource=building_data.get("BuildResource") or building_data.get("2").get("BuildResource")
+                ),
+                "village": "home" if not village_type else "builderBase",
+                "width": building_data.get("Width", 1), #walls are null for some reason, so let's make it 1
+                "superchargeable": superchargeable,
+                "levels": []
+            }
+
+            # put seasonal defense onto the crafting station
+            if _id == 1000097:
```

Comment:
so this is completely dependent on the position in the apk data and would break whenever it would move? can we not define a position independent criteria for this?